### ISSUE TITLE
[SCM-886] Tests with checkin rely on global Git config

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1123]
**[Matthieu Brouillard](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcfoggy)** opened **[SCM-886](https://issues.apache.org/jira/browse/SCM-886?redirect=false)** and commented

Inside maven-scm-provider-gitexe project, the two following tests `GitCheckInCommandNoBranchTest.testCheckinNoBranch` and `GitCheckInCommandTest.testCheckinAfterRename` expect a global user.name and user.email to be set.

On installations where .gitconfig contains the following:

`[user]`
`useConfigOnly = true`

The tests will fail.

It would be better after repository creation to configure a user name & email at project level.

---

**Affects:** 1.9.5

**Issue Links:**
- [SCM-905](https://issues.apache.org/jira/browse/SCM-905) Tests with checkin rely on global Git config (2)
 (_**"is cloned by"**_)

**Remote Links:**
- [GitHub Pull Request #73
](https://github.com/apache/maven-scm/pull/73)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I added this param to my global config and cannot reproduce it on Windows in PowerShell. Any advises?

**Comment by jira-importer** [comment]
**[Matthieu Brouillard](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcfoggy)** commented

Make sure that you also do not have already launched a `git config user.email` and `git config user.name` on the working repository.

I saw the error because on my setup I have globally

[user]
useConfigOnly = true
name = Matthieu Brouillard

Notice that the `user.email` is missing globally. This prevents me to commit on repositories without configuring correctly the email I want for the particular repository. So in order to commit to any repository, I need to do a `git config user.email` at the repository level ; which I had not done on maven-scm project and that implied some tests to fail (then ones doing commits).

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I can reproduce this now:

```
Failed tests:
  GitExeBlameCommandTckTest.testBlameCommand:44->BlameCommandTckTest.testBlameCommand:73 Unable to checkin changes to the repository
  GitExeBranchCommandTckTest>BranchCommandTckTest.testBranchCommandTest:73->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitExeChangeLogCommandTckTest>ChangeLogCommandTckTest.testChangeLogCommand:76 Unable to checkin changes to the repository
  GitCheckInCommandNoBranchTest.testCheckinNoBranch:75->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitCheckInCommandTest.testCheckinAfterRename:113->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitExeCheckInCommandTckTest>CheckInCommandTckTest.testCheckInCommandPartialFileset:149->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitExeCheckInCommandTckTest>CheckInCommandTckTest.testCheckInCommandTest:78->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitExeStatusCommandTckTest>StatusCommandTckTest.testStatusCommand:102->StatusCommandTckTest.commit:64 Check result was successful, output:
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: no email was given and auto-detection is disabled

  GitExeTagCommandTckTest>TagCommandTckTest.testTagCommandTest:58->ScmTestCase.assertResultIsSuccess:208 The check out result success flag was false.
  GitExeUpdateCommandTckTest>UpdateCommandTckTest.testUpdateCommand:131->UpdateCommandTckTest.commit:67 Check result was successful, output:
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: no email was given and auto-detection is disabled


Tests run: 82, Failures: 10, Errors: 0, Skipped: 1
```

Do you think you could provide a fix? I guess a static `.git/config` in the test repo shall be enough.

**Comment by jira-importer** [comment]
**[Matthieu Brouillard](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mcfoggy)** commented

> I guess a static `.git/config` in the test repo shall be enough.

No it is not because the clone command already created a config file. We could write/append to it but it is quite ugly ; the best is to issue a git config command on this cloned repository. But I have not seen any command allowing to call 'config', did I miss looked?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The clone in question is `D:\Entwicklung\Projekte\maven-scm\maven-scm-providers\maven-scm-providers-git\maven-scm-provider-gitexe\target\checkin-nobranch\.git\config`.
Unfortunately, there is no `config` command because it is too Git specific. The best you can do in the tests `GitCheckInCommandNoBranchTest.java` and `GitExeCheckOutCommandNoBranchTest.java` is to issue a exe command with the given wrappers. It won't get better.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [7719468daf9432e88c34005ca6054f6c07d3f4b3](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=7719468daf9432e88c34005ca6054f6c07d3f4b3).
